docs/src/mock.md

```diff
@@ -3,121 +3,524 @@ id: mock
+
+The `update` option updates the given HAR file with the actual network information or creates a new HAR file, in the hars folder, if one doesn't already exist. In order to record the HAR file, set `update` to true.
+
+```js
+test('records or updates the HAR file', async ({
+  page,
+}) => {
+  // Get the response and add to it
+  await page.routeFromHAR('./hars/fruit.har', {
+    url: '*/**/api/v1/fruits',
+    update: true,
+  });
+  // Go to the page
+  await page.goto('https://demo.playwright.dev/api-mocking');
+  // Assert that the fruit is visible
+  await expect(page.getByText('Strawberry')).toBeVisible();
+});
+```
+
+```python async
+async def records_or_updates_the_har_file(page):
+  # Get the response from the HAR file
+  await page.route_from_har("./hars/fruit.har", url="*/**/api/v1/fruits", update=True)
+
+  # Go to the page
+  await page.goto("https://demo.playwright.dev/api-mocking")
+
+  # Assert that the Playwright fruit is visible
+  await page.get_by_text("Strawberry").to_be_visible()
+```
+
+```python sync
+def records_or_updates_the_har_file(page):
+  # Get the response from the HAR file
+  page.route_from_har("./hars/fruit.har", url="*/**/api/v1/fruits", update=True)
+
+  # Go to the page
+  page.goto("https://demo.playwright.dev/api-mocking")
+
+  # Assert that the Playwright fruit is visible
+  page.get_by_text("Strawberry").to_be_visible()
+```
+
+```csharp
+await page.RouteFromHARAsync(
+  "./hars/fruit.har",
+  new RouteFromHAROptions
+  {
+    Url = "*/**/api/v1/fruits",
+    Update = true,
+  }
+);
+
+// Go to the page
+await page.GotoAsync("https://demo.playwright.dev/api-mocking");
+
+// Assert that the Playwright fruit is visible
+await page.ExpectByTextAsync("Strawberry").ToBeVisibleAsync();
+```
+
+```java
+page.routeFromHAR("./hars/fruit.har", new RouteFromHAROptions()
+  .setUrl("*/**/api/v1/fruits")
+  .setUpdate(true)
+);
+
+// Go to the page
+page.goto("https://demo.playwright.dev/api-mocking");
+
+// Assert that the Playwright fruit is visible
+assertThat(page.getByText("Strawberry")).isVisible();
+```
+
+### Modifying a HAR file
+
+Once you have recorded a HAR file you can modify it by opening the hashed .txt file inside your 'hars' folder and editing the JSON. This file should be committed to your source control. Anytime you run this test with `update: true` it will update your HAR file with the request from the API. 
+
+```json
+[
+  {
+    "name": "Playwright",
+    "id": 100
+  },
+  // ... other fruits
+]
+```
+
+### Replaying from HAR
+
+Use [`method: Page.routeFromHAR`] or [`method: BrowserContext.routeFromHAR`] to serve matching responses from the [HAR](http://www.softwareishard.com/blog/har-12-spec/) file. Set `update: false` as an option in the `routeFromHAR` method or simply remove this option as the default is false. This will run the test against the HAR file instead of hitting the API.
```

Comment:
```suggestion
Now that you have the HAR file recorded and modified the mock data, it can be used to serve matching responses in the test. For this, just turn off or simply remove the `update` option. This will run the test against the HAR file instead of hitting the API.
```